packages/providers/src/provider.ts

```diff
@@ -279,17 +282,22 @@ export default class Provider {
     timeout: undefined,
     cacheUtxo: undefined,
     fetch: undefined,
+    retryOptions: undefined,
   };
 
-  private static getFetchFn(options: ProviderOptions) {
-    return options.fetch !== undefined
-      ? options.fetch
-      : (url: string, request: FetchRequestOptions) =>
-          fetch(url, {
-            ...request,
-            signal:
-              options.timeout !== undefined ? AbortSignal.timeout(options.timeout) : undefined,
-          });
+  private static getFetchFn(options: ProviderOptions): NonNullable<ProviderOptions['fetch']> {
+    return retrier((...args) => {
+      if (options.fetch) {
+        return options.fetch(...args);
+      }
+
+      const url = args[0];
+      const request = args[1];
+      return fetch(url, {
+        ...request,
+        signal: options.timeout !== undefined ? AbortSignal.timeout(options.timeout) : undefined,
+      });
+    }, options.retryOptions);
```

Comment:
Suggestion on top of https://github.com/FuelLabs/fuels-ts/pull/1474#discussion_r1434329966.

```suggestion
    const { retryOptions, timeout } = options;

    return retrier((...args) => {
      if (options.fetch) {
        return options.fetch(...args);
      }

      const url = args[0];
      const request = args[1];
      const signal = timeout ? AbortSignal.timeout(timeout) : undefined;

      return fetch(url, { ...request, signal });
    }, retryOptions);
```